### DISCUSSION TITLE
how to get the node surrounding the cursor?

### DISCUSSION THREAD

**Discussion by philippehenri** [Alex-D/Trumbowyg#1351] [Q&A] [answered]
Hello,

I have a text:

aaaa **_bbb_** ccc

I would like to know if it is possible to get the span surrounding the cursor (when a left or right key is pressed for example)?

thanks
Philippe

**Comment by Alex-D** [comment, marked as answer]
Try something like:
```js
let selectedNode = document.getSelection().focusNode

while (['SPAN'].indexOf(selectedNode.nodeName) < 0) { 
     selectedNode = selectedNode.parentNode; 
}

// selectedNode contains the first parent node you want
```

Inspired from:
https://github.com/Alex-D/Trumbowyg/blob/f5f921cc85176e8f1b6c5620571db38e80ed7bc0/src/trumbowyg.js#L1289-L1313

**Comment by philippehenri** [comment]
thanks Alex !
